app/src/main/java/org/mozilla/focus/utils/IntentUtils.java

```diff
@@ -84,6 +84,8 @@ public static boolean handleExternalUri(final Context context, final String uri)
 
         if (matchingActivities.size() > 0) {
             context.startActivity(intent);
+        } else {
+            return false;
```

Comment:
We agreed on not handling unknown urls so I don't think we should flip this. (I don't think we need to flip this fo this feature, either.)

Reply:
How about let's make the false return as an error handling case?
```
if (matchingActivities <= 0) {
  return false;
}

// return true when handled with startActivity
context.startActivity(intent);
return true;
```

Reply:
It was our earlier decision to bypass non-supported link like pinterest://, see c38807ae01e27402c6d6055d4b588295374ec07c. However I agree the current naming is misleading so I would suggest splitting these into to functions shouldHandleExternal and canHandleExternal.

Reply:
Also, I agree that in canHandleExternal, the style @mxlius has provided is clearer.

Reply:
since we have one more criteria "ignore" now, I've used a different approach:
```
if (matchingActivities.size() > 0) {
    // only start the activity when there's at least one handler so don't need to catch ActivityNotFoundException here.
    context.startActivity(intent);
    return true;
} else {
    // there's no one can handle this url, return true if it's okay or false otherwise.
    return ignore;
}
```